src/time.rs

```diff
@@ -73,8 +72,8 @@ impl Into<KiloHertz> for MegaHertz {
 
 impl From<u32> for Hertz {
     fn from(ms: u32) -> Self {
-        if ms >= 1000 {
-            Hertz(1/(ms/1000))
+        if ms <= 1000 {
+            Hertz((1_f32/((ms as f32)/1000_f32)) as u32)
```

Comment:
The rounding can be implemented in FPA, like so:

```suggestion
            Hertz((1000 + ms / 2) / ms)
```

Reply:
Well, thats quite a lot cleaner :)